### PR TITLE
Use MySQL 8.0.28 on CircleCI

### PR DESCRIPTION
**Pull request by zenovich**
As we have MySQL 8.0.28 in prod, it would be a good idea to run our tests against it.

### CODE REVIEW

**Comment by zenovich** [comment]
Oops! It looks like our project is not fully compatible with the MySQL in prod :(

 @smadbe , @GeoffreyHuck

**Comment by zenovich** [comment]
The root issue:
```
Unable to apply migration: Error 3995: Character set 'binary' cannot be used in conjunction with 'utf8mb4_0900_ai_ci' in call to regexp_like. handling 2002121859_create_tables_for_batches_of_users.sql
make: *** [Makefile:76: db-migrate] Error 1
```
I'll try to fix it

**Comment by zenovich** [comment]
Also, I'll try to create a migration fixing the prod.

**Comment by zenovich** [comment]
The issue is related to user_batches table. It contains a check constraint 
```
CONSTRAINT `ck_user_batches_custom_prefix` CHECK (regexp_like(cast(`custom_prefix` as char charset binary),_utf8mb4'^[a-z0-9-]+$'))
```
which works fine on MySQL < 8.0.22, but is completely incompatible with MySQL 8.0.22+. After upgrading MySQL to 8.0.22+ (8.0.28 in our case), this table became inaccessible: it's impossible to select data from it, alter it, rename it, drop it:

```
mysql> select * from `user_batches`;
ERROR 3995 (HY000): Character set 'binary' cannot be used in conjunction with 'utf8mb4_0900_ai_ci' in call to regexp_like.
mysql> ALTER TABLE `user_batches` DROP CHECK `ck_user_batches_custom_prefix`;
ERROR 3995 (HY000): Character set 'binary' cannot be used in conjunction with 'utf8mb4_0900_ai_ci' in call to regexp_like.
mysql> show create table `user_batches`;
ERROR 3995 (HY000): Character set 'binary' cannot be used in conjunction with 'utf8mb4_0900_ai_ci' in call to regexp_like.
mysql> drop table `user_batches`;
ERROR 3995 (HY000): Character set 'binary' cannot be used in conjunction with 'utf8mb4_0900_ai_ci' in call to regexp_like.
```

The only way to deal with this is to create another table and use it instead of user_batches.

So here we:
1. Change MySQL version to 8.0.28 in .circleci/config.yml (for tests and migration-tests)
2. Patch the large dump of data before seeding the database, otherwise it fails on loading user_batches table.
3. Make the same fixes as in #1113 (create table user_batches_new with the same structure as of user_batches, fix old migrations related to user_batches to make them compatible with MySQL 8.0.28, replace user_batches with user_batches_new everywhere in the code).

**Review by smadbe — APPROVED** [review]

File: db/migrations/2407171717_create_table_user_batches_new.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-- +migrate Up
+
+CREATE TABLE IF NOT EXISTS `user_batches_new` (
```
[the remark this answers is not in the document]
Comment (zenovich) [reply db/migrations/2407171717_create_table_user_batches_new.sql]: fixed

**Pull request merged**